java-spiffe-provider/src/main/java/spiffe/provider/SpiffeTrustManagerFactory.java

```diff
@@ -0,0 +1,118 @@
+ * @see SpiffeSslContextFactory
+ */
+public class SpiffeTrustManagerFactory extends TrustManagerFactorySpi {
+
+    /**
+     * Default method for creating a TrustManager initializing it with
+     * the {@link spiffe.workloadapi.X509Source} instance
+     * that is handled by the {@link X509SourceManager}, and
+     * with and a supplier of accepted SPIFFE IDs. that reads the list
+     * from the System Property defined in SSL_SPIFFE_ACCEPT_PROPERTY.
+     *
+     * @return a TrustManager array with an initialized TrustManager.
+     * @throws SpiffeProviderException in case there is an error setting up the X509 source
+     */
+    @Override
+    public TrustManager[] engineGetTrustManagers() {
+        SpiffeTrustManager spiffeTrustManager =
+                null;
+        try {
+            spiffeTrustManager = new SpiffeTrustManager(
+                    X509SourceManager.getX509Source(),
+                    this::getAcceptedSpiffeIds
+            );
+        } catch (X509SourceException e) {
+            throw new SpiffeProviderException("The X509 source could not be created", e);
+        } catch (SocketEndpointAddressException e) {
+            throw new SpiffeProviderException("The Workload API Socket endpoint address configured is not valid", e);
+        }
+        return new TrustManager[]{spiffeTrustManager};
+    }
+
+    /**
+     * Creates a TrustManager initializing it with the X509BundleSource instance
+     * and with and a supplier of accepted SPIFFE IDs. that reads the list
+     * from the System Property defined in SSL_SPIFFE_ACCEPT_PROPERTY.
+     *
+     * @return a TrustManager array with an initialized TrustManager.
+     */
+    public TrustManager[] engineGetTrustManagers(X509BundleSource x509BundleSource) {
+        val spiffeTrustManager =
+                new SpiffeTrustManager(
+                        x509BundleSource,
+                        this::getAcceptedSpiffeIds
+                );
+        return new TrustManager[]{spiffeTrustManager};
+    }
+
+    /**
+     * Creates a TrustManager initializing it with a X509BundleSource to get the bundles,
+     * with a function verify a chain of certificates using a to validate the SPIFFE IDs
+     * of the peer's certificates, and a supplier of accepted SPIFFE IDs.
+     *
+     * @param x509BundleSource a {@link X509BundleSource} to provide the X.509-Bundles
+     * @param acceptedSpiffeIdsSupplier a Supplier to provide a List of SPIFFE IDs that are accepted
+     * @return a TrustManager array with an initialized TrustManager.
+     */
+    public TrustManager[] engineGetTrustManagers(
+            X509BundleSource x509BundleSource,
+            Supplier<List<SpiffeId>> acceptedSpiffeIdsSupplier) {
+
+        Supplier<List<SpiffeId>> spiffeIdsSupplier;
+        if (acceptedSpiffeIdsSupplier != null) {
+            spiffeIdsSupplier = acceptedSpiffeIdsSupplier;
+        } else {
+            spiffeIdsSupplier = this::getAcceptedSpiffeIds;
+        }
```

Comment:
Would it make sense to you to use this form instead:
```suggestion
        final Supplier<List<SpiffeId>> spiffeIdsSupplier = 
        		acceptedSpiffeIdsSupplier != null ? 
        		acceptedSpiffeIdsSupplier : this::getAcceptedSpiffeIds;

```
Personally, I found it easier to read for this case.

Reply:
Done.